src/compiler/transformers/esDecorators.ts

```diff
@@ -1072,8 +1072,8 @@ export function transformESDecorators(context: TransformationContext): (x: Sourc
             if (initializerStatements) {
                 const statements: Statement[] = [];
                 const nonPrologueStart = factory.copyPrologue(node.body.statements, statements, /*ensureUseStrict*/ false, visitor);
-                addRange(statements, initializerStatements);
                 addRange(statements, visitNodes(node.body.statements, visitor, isStatement, nonPrologueStart));
+                addRange(statements, initializerStatements);

```

Comment:
This isn't the correct fix, as it would have instance initializers run after all other statements in the body.

Reply:
Maybe it's out of scope for this PR - but why don't we just transform all `super()` calls, stashing away the initializer statements as we encounter nested classes instead of trying to find the index?